modin/core/dataframe/base/partitioning/axis_partition.py

```diff
@@ -0,0 +1,161 @@
+            A list of `BaseDataframePartition` objects.
+
+        Notes
+        -----
+        The procedures that invoke this method assume full axis
+        knowledge. Implement this method accordingly.
+
+        You must return a list of `BaseDataframePartition` objects from this method.
+        """
+        pass
+
+    def shuffle(self, func, lengths, **kwargs):
+        """
+        Shuffle the order of the data in this axis partition based on the `lengths`.
+
+        Parameters
+        ----------
+        func : callable
+            The function to apply before splitting.
+        lengths : list
+            The list of partition lengths to split the result into.
+        **kwargs : dict
+            Additional keywords arguments to be passed in `func`.
+
+        Returns
+        -------
+        list
+            A list of `BaseDataframePartition` objects split by `lengths`.
+        """
+        pass
+
+    # Child classes must have these in order to correctly subclass.
+    instance_type = None
+    partition_type = None
+
+    def _wrap_partitions(self, partitions):
+        """
+        Wrap remote partition objects with `BaseDataframePartition` class.
+
+        Parameters
+        ----------
+        partitions : list
+            List of remotes partition objects to be wrapped with `BaseDataframePartition` class.
+
+        Returns
+        -------
+        list
+            List of wrapped remote partition objects.
+        """
+        return [self.partition_type(obj) for obj in partitions]
+
+    def force_materialization(self, get_ip=False):
+        """
+        Materialize axis partitions into a single partition.
+
+        Parameters
+        ----------
+        get_ip : bool, default: False
+            Whether to get node ip address to a single partition or not.
+
+        Returns
+        -------
+        BaseDataframeAxisPartition
+            An axis partition containing only a single materialized partition.
+        """
+        materialized = self.apply(
+            lambda x: x, num_splits=1, maintain_partitioning=False
+        )
+        return type(self)(materialized, get_ip=get_ip)
+
+    def unwrap(self, squeeze=False, get_ip=False):
+        """
+        Unwrap partitions from this axis partition.
+
+        Parameters
+        ----------
+        squeeze : bool, default: False
+            Flag used to unwrap only one partition.
+        get_ip : bool, default: False
+            Whether to get node ip address to each partition or not.
+
+        Returns
+        -------
+        list
+            List of partitions from this axis partition.
+
+        Notes
+        -----
+        If `get_ip=True`, a list of tuples of Ray.ObjectRef/Dask.Future to node ip addresses and
+        unwrapped partitions, respectively, is returned if Ray/Dask is used as an engine
+        (i.e. [(Ray.ObjectRef/Dask.Future, Ray.ObjectRef/Dask.Future), ...]).
+        """
+        if squeeze and len(self.list_of_blocks) == 1:
```

Comment:
same, it's an abstract base, why implement anything in it?

Reply:
It's legacy but yes we want to keep abstract base classes with methods declaration only. Should probably be fixed as part of a separate issue.